org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/LilyPondBuilder.java

```diff
@@ -131,8 +140,60 @@ public int compare(IFile o1, IFile o2) {
 				return o1.getName().compareTo(o2.getName());
 			}
 		});
+		if(!continueWithOpenFilesDirty(sortedFiles)) {
+			return ImmutableList.of();
+		}
 		return sortedFiles;
+	}
 
+	private boolean continueWithOpenFilesDirty(Collection<IFile> filesToCompile) {
+		AtomicBoolean doContinue=new AtomicBoolean(true);
+		List<IFile> allOpenDirtyFiles=LilyPondPerspective.getAllOpenDirtyFiles();
+		if(!allOpenDirtyFiles.isEmpty()) {
+			List<String> openDirtyFilesToCompile=new ArrayList<>();
+			Set<IFile> filesInvolvedInCompilation=new HashSet<>(filesToCompile);
+			filesInvolvedInCompilation.addAll(getTransitivelyIncludedFiles(filesToCompile));
+			for (IFile openDirtyFile : allOpenDirtyFiles) {
+				if(filesInvolvedInCompilation.contains(openDirtyFile)) {
+					openDirtyFilesToCompile.add(openDirtyFile.getName());
+				}
+			}
+			if(!openDirtyFilesToCompile.isEmpty()) {
+				Display.getDefault().syncExec(new Runnable() {
+					@Override
+					public void run() {
+						doContinue.set(MessageDialog.openQuestion(PlatformUI.getWorkbench().getDisplay().getActiveShell(),
+								"Continue compilation ignoring unsaved changes",
+								"The compilation involves unsaved files opened in an editor:\n"+
+										Joiner.on(", ").join(openDirtyFilesToCompile)+
+								"\n\nUnsaved changes made to these files are ignored in the compailation. \nDo you want to continue?"));
```

Comment:
Fix the typo: compailation -> compilation

Reply:
done